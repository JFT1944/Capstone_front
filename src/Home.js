import React from "react";
import { NavLink } from "react-router-dom";



function Home(props){
let {user, token} = props
console.log(user)


if(user){
    return(
        <>
        <div style={{width:'100vw', height:'100vh', display:'flex', justifyContent:'center', alignItems:'center', flexDirection:'column', color:'#0F683A'}}>
        <h2>Welcome {user.firstName}</h2>
        </div>
        </>
    )
}else{
    return(
        <>
        <div style={{width:'100vw', height:'100vh', display:'flex', justifyContent:'center', alignItems:'center', flexDirection:'column', color:'#0F683A'}}>
        <h2>Welcome to Dispensa<span className="logo2">.io!</span></h2>
        <h3>The best place to store your digital pantry and speed up your shopping experience.</h3>
         <NavLink exact to='/login' >Login:</NavLink>
         <NavLink exact to='/signup' >Sign Up:</NavLink>
        </div>
        </>
    )
}
    
}

export default Home