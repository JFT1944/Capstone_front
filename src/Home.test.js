
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import  Home from "./Home";
import { MemoryRouter } from "react-router-dom";


// write a smoke test for the home component

it('should appear in a smoketest', () => {
    render(<Home user={'user'} token={'token'}/>)
})


it('should match snapshot', () => {
    const {asFragment} = render(<MemoryRouter><Home user={'user'} setUser={'setUser'} token={'token'} setToken={'setToken'}/></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})