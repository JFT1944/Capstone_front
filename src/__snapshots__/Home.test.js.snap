// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100vw; height: 100vh; display: flex; justify-content: center; align-items: center; flex-direction: column; color: rgb(15, 104, 58);"
  >
    <h2>
      Welcome 
    </h2>
  </div>
</DocumentFragment>
`;
